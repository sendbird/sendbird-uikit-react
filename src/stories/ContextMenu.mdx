import { Meta, Canvas, Controls } from '@storybook/addon-docs';
import ContextMenu, { MenuItems, MenuItem } from '../ui/ContextMenu';
import * as ContextMenuStories from './ContextMenu.stories';

<Meta title="UI/ContextMenu" component={ContextMenu} of={ContextMenuStories} />

# ContextMenu

The `ContextMenu` component allows for dynamic menu creation triggered by user actions. It provides a flexible way to present a list of interactive options to the user.

<Controls />

### Importing ContextMenu

To integrate the ContextMenu component into your application, use the following import statement:

```tsx
import ContextMenu, { MenuItems, MenuItem } from '@sendbird/uikit-react/ui/ContextMenu';
```
