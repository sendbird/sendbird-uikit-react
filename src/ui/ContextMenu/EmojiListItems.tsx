import React, { ReactElement, ReactNode, RefObject, useEffect, useRef, useState } from 'react';
import { createPortal } from 'react-dom';

import SortByRow from '../SortByRow';

type SpaceFromTrigger = { x: number, y: number };
type ReactionStyle = { left: number, top: number };
export interface EmojiListItemsProps {
  closeDropdown: () => void;
  children: ReactNode;
  parentRef: RefObject<HTMLDivElement> | null;
  parentContainRef: RefObject<HTMLDivElement>;
  spaceFromTrigger?: SpaceFromTrigger;
}

const EmojiListItems = ({
  children,
  parentRef,
  parentContainRef,
  spaceFromTrigger = { x: 0, y: 0 },
  closeDropdown,
}: EmojiListItemsProps): ReactElement => {
  const [reactionStyle, setReactionStyle] = useState<ReactionStyle>({ left: 0, top: 0 });
  const reactionRef = useRef(null);

  /* showParent & hideParent */
  useEffect(() => {
    if (parentContainRef && parentContainRef?.current) {
      parentContainRef.current.classList.add('sendbird-reactions--pressed');
    }
    return () => {
      if (parentContainRef && parentContainRef?.current) {
        parentContainRef.current.classList.remove('sendbird-reactions--pressed');
      }
    };
  }, []);

  /* setupEvents & cleanupEvents */
  useEffect(() => {
    const handleClickOutSide = (event) => {
      if (reactionRef?.current && !reactionRef?.current?.contains?.(event.target)) {
        closeDropdown();
      }
    };
    document.addEventListener('mousedown', handleClickOutSide);
    return () => {
      document.removeEventListener('mousedown', handleClickOutSide);
    };
  }, []);

  /* getBarPosition */
  useEffect(() => {
    const spaceFromTriggerX = spaceFromTrigger?.x || 0;
    const spaceFromTriggerY = spaceFromTrigger?.y || 0;
    const parentRect = parentRef?.current?.getBoundingClientRect();
    const x = parentRect.x || parentRect.left;
    const y = parentRect.y || parentRect.top;
    const reactionStyle = {
      top: y,
      left: x,
    };
    if (!reactionRef?.current) {
      setReactionStyle(reactionStyle);
    } else {
      const rect = reactionRef?.current?.getBoundingClientRect();
      if (reactionStyle.top < rect.height) {
        reactionStyle.top += parentRect.height;
        reactionStyle.top += spaceFromTriggerY;
      } else {
        reactionStyle.top -= rect.height;
        reactionStyle.top -= spaceFromTriggerY;
      }
      reactionStyle.left -= rect.width / 2;
      reactionStyle.left += (parentRect.height / 2) - 2;
      reactionStyle.left += spaceFromTriggerX;
      const maximumLeft = window.innerWidth - rect.width;
      if (maximumLeft < reactionStyle.left) {
        reactionStyle.left = maximumLeft;
      }
      if (reactionStyle.left < 0) {
        reactionStyle.left = 0;
      }
      setReactionStyle(reactionStyle);
    }
  }, []);

  return (
    createPortal(
      <>
        <div className="sendbird-dropdown__menu-backdrop" />
        <ul
          className="sendbird-dropdown__reaction-bar"
          ref={reactionRef}
          style={{
            display: 'inline-block',
            position: 'fixed',
            left: `${Math.round(reactionStyle.left)}px`,
            top: `${Math.round(reactionStyle.top)}px`,
          }}
        >
          <SortByRow
            className="sendbird-dropdown__reaction-bar__row"
            maxItemCount={8}
            itemWidth={44}
            itemHeight={40}
          >
            {children}
          </SortByRow>
        </ul>
      </>,
      document.getElementById('sendbird-emoji-list-portal'),
    )
  );
};

export default EmojiListItems;
