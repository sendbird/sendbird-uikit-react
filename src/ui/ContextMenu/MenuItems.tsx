import React, { ReactElement } from 'react';
import { createPortal } from 'react-dom';

interface MenuItemsProps {
  className?: string;
  style?: Record<string, string>;
  openLeft?: boolean;
  children: React.ReactElement | Array<React.ReactElement> | React.ReactNode;
  parentRef: React.RefObject<HTMLElement>;
  parentContainRef: React.RefObject<HTMLElement>;
  closeDropdown: () => void;
}

type MenuStyleType = { top?: number, left?: number };
interface MenuItemsState {
  menuStyle: MenuStyleType;
  handleClickOutside: (e: MouseEvent) => void;
}

// padding to handle height of last item in message-list
const HEIGHT_PADDING = 60;

export default class MenuItems extends React.Component<MenuItemsProps, MenuItemsState> {
  constructor(props: MenuItemsProps) {
    super(props);
    this.state = {
      menuStyle: {},
      handleClickOutside: () => { /* noop */ },
    };
  }

  menuRef: React.RefObject<HTMLUListElement> = React.createRef();

  componentDidMount(): void {
    this.setupEvents();
    this.getMenuPosition();
  }

  componentWillUnmount(): void {
    this.cleanUpEvents();
  }

  setupEvents = (): void => {
    const { closeDropdown } = this.props;
    const { menuRef } = this;
    const handleClickOutside = (event) => {
      if (menuRef?.current && !menuRef?.current?.contains?.(event.target)) {
        closeDropdown?.();
      }
    };
    this.setState({
      handleClickOutside,
    });

    document.addEventListener('mousedown', handleClickOutside);
  };

  cleanUpEvents = (): void => {
    const {
      handleClickOutside,
    } = this.state;
    document.removeEventListener('mousedown', handleClickOutside);
  };

  getMenuPosition = (): MenuStyleType => {
    const { parentRef, openLeft } = this.props;
    const parentRect = parentRef?.current?.getBoundingClientRect?.();
    const x = parentRect?.x || parentRect?.left || 0;
    const y = parentRect?.y || parentRect?.top || 0;
    const menuStyle = {
      top: y,
      left: x,
    };
    if (!this.menuRef.current) return menuStyle;
    const { innerWidth, innerHeight } = window;
    const rect = this.menuRef.current.getBoundingClientRect();
    if (y + rect.height + HEIGHT_PADDING > innerHeight) {
      menuStyle.top -= rect.height;
    }
    if (x + rect.width > innerWidth && !openLeft) {
      menuStyle.left -= rect.width;
    }
    if (menuStyle.top < 0) {
      menuStyle.top = rect.height < innerHeight ? (innerHeight - rect.height) / 2 : 0;
    }
    menuStyle.top += 32;
    if (openLeft) {
      const padding = Number.isNaN(rect.width - 30)
        ? 108 // default
        : rect.width - 30;
      menuStyle.left -= padding;
    }
    // warning: this section has to be executed after the openLeft is calculated
    // menu is outside viewport
    if (menuStyle.left < 0) {
      menuStyle.left = rect.width < innerWidth ? (innerWidth - rect.width) / 2 : 0;
    }

    this.setState({ menuStyle });
    return menuStyle;
  };

  render(): ReactElement {
    const { menuStyle } = this.state;
    const { children, style, className = '' } = this.props;
    return (
      createPortal(
        (
          <div className={this.props?.className}>
            <div className="sendbird-dropdown__menu-backdrop" />
            <ul
              className={`${className} sendbird-dropdown__menu`}
              ref={this.menuRef}
              style={{
                display: 'inline-block',
                position: 'fixed',
                left: `${Math.round(menuStyle.left)}px`,
                top: `${Math.round(menuStyle.top)}px`,
                ...style,
              }}
            >
              {children}
            </ul>
          </div>
        ),
        document.getElementById('sendbird-dropdown-portal'),
      )
    );
  }
}
