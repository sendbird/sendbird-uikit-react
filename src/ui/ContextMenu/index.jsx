import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './index.scss';

import Label, { LabelTypography, LabelColors } from '../Label';
import { getClassName } from '../../utils';

import MenuItems_ from './items/MenuItems';
import EmojiListItems_ from './items/EmojiListItems';

const ENTER = 13;

export const MenuItems = MenuItems_;
export const EmojiListItems = EmojiListItems_;

export const MenuItem = ({
  className,
  children,
  onClick,
  disable,
}) => {
  const handleClickEvent = (e) => { if (!disable) onClick(e); };
  return (
    <li
      className={getClassName([className, 'sendbird-dropdown__menu-item', disable ? 'disable' : ''])}
      role="menuitem"
      onClick={handleClickEvent}
      onKeyPress={(e) => { if (e.keyCode === ENTER) handleClickEvent(e); }}
      tabIndex={0}
    >
      <Label
        className="sendbird-dropdown__menu-item__text"
        type={LabelTypography.SUBTITLE_2}
        color={disable ? LabelColors.ONBACKGROUND_4 : LabelColors.ONBACKGROUND_1}
      >
        {children}
      </Label>
    </li>
  );
};

MenuItem.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
  ]).isRequired,
  onClick: PropTypes.func.isRequired,
  disable: PropTypes.bool,
};

MenuItem.defaultProps = {
  className: '',
  disable: false,
};

// Root components should be appended before ContextMenu is rendered
export const MenuRoot = () => (
  <div id="sendbird-dropdown-portal" />
);
export const EmojiReactionListRoot = () => (
  <div id="sendbird-emoji-list-portal" />
);

export default function ContextMenu({ menuTrigger, menuItems }) {
  const [showMenu, setShowMenu] = useState(false);

  return (
    <div className="sendbird-context-menu" style={{ display: 'inline' }}>
      {menuTrigger(() => setShowMenu(!showMenu))}
      {showMenu && menuItems(() => setShowMenu(false))}
    </div>
  );
}

ContextMenu.propTypes = {
  menuTrigger: PropTypes.func.isRequired,
  menuItems: PropTypes.func.isRequired,
};
