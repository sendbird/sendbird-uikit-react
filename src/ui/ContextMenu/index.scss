@import '../../styles/variables';

.sendbird__offline {
  .sendbird-dropdown__menu .sendbird-dropdown__menu-item {
    cursor: not-allowed;
  }
}

.sendbird-dropdown__menu {
  z-index: 99999;
  position: absolute;
  top: 100%;
  left: 0;
  min-width: 140px;
  margin: 0px;
  padding: 8px 0px;
  border-radius: 4px;
  list-style: none;
  @include themed() {
    box-shadow: t(shadow-gradient-2);
    background-color: t(bg--0-5);
  }

  .sendbird-dropdown__menu-item {
    cursor: pointer;
    white-space: nowrap;
    padding: 8px 16px;
    &.disable {
      cursor: not-allowed;
      &:hover {
        @include themed() {
          background-color: t(bg--0-5);
        }
      }
    }

    &:hover {
      @include themed() {
        background-color: t(bg--1-4);
      }
    }
  }
}

.sendbird-dropdown__reaction-bar {
  position: absolute;
  top: 100%;
  left: 0;
  min-width: 44px;
  max-width: 352px;
  max-height: 208px;
  overflow-y: scroll;
  margin: 0px;
  padding: 8px;
  border-radius: 8px;
  list-style: none;

  @include themed() {
    box-shadow: t(shadow-gradient-2);
    background-color: t(bg--0-5);
  }

  .sendbird-dropdown__reaction-bar__row {
    .sendbird-reaction-button, .sendbird-reaction-button--selected {
      margin: 4px;
    }
  }
}

.sendbird-dropdown__menu-backdrop {
  background-color: transparent;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.sendbird-reactions--pressed {
  display: block !important;
}
.sendbird-icon--pressed {
  display: block !important;
}
