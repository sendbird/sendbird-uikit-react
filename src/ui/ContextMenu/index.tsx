import React, { ReactElement, MouseEvent, useState, ReactNode } from 'react';
import './index.scss';

import _MenuItems from './MenuItems';
import _EmojiListItems from './EmojiListItems';

import { getClassName } from '../../utils';
import Label, { LabelTypography, LabelColors } from '../Label';

const ENTER_KEY = 13;

export const MenuItems = _MenuItems;
export const EmojiListItems = _EmojiListItems;

export interface MenuItemProps {
  className?: string | Array<string>;
  children: ReactNode;
  onClick?: (e: MouseEvent<HTMLLIElement>) => void;
  disable?: boolean;
}
export const MenuItem = ({
  className = '',
  children,
  onClick,
  disable = false,
}: MenuItemProps): ReactElement => {
  const handleClickEvent = (e) => {
    if (!disable && onClick) {
      onClick?.(e);
    }
  };
  return (
    <li
      className={getClassName([className, 'sendbird-dropdown__menu-item', disable ? 'disable' : ''])}
      role="menuitem"
      onClick={handleClickEvent}
      onKeyPress={(e) => { if (e.keyCode === ENTER_KEY) handleClickEvent(e); }}
      tabIndex={0}
    >
      <Label
        className="sendbird-dropdown__menu-item__text"
        type={LabelTypography.SUBTITLE_2}
        color={disable ? LabelColors.ONBACKGROUND_4 : LabelColors.ONBACKGROUND_1}
      >
        {children}
      </Label>
    </li>
  );
};

export const MenuRoot = (): ReactElement => (
  <div
    id="sendbird-dropdown-portal"
    className="sendbird-dropdown-portal"
  />
);
export const EmojiReactionListRoot = (): ReactElement => <div id="sendbird-emoji-list-portal" />;

type MenuDisplayingFunc = () => void;
export interface ContextMenuProps {
  menuTrigger: (MenuDisplayingFunc) => ReactElement;
  menuItems: (MenuDisplayingFunc) => ReactElement;
}
export default function ContextMenu({
  menuTrigger,
  menuItems,
}: ContextMenuProps): ReactElement {
  const [showMenu, setShowMenu] = useState(false);
  return (
    <div
      className="sendbird-context-menu"
      style={{ display: 'inline' }}
    >
      {menuTrigger(() => setShowMenu(!showMenu))}
      {showMenu && menuItems(() => setShowMenu(false))}
    </div>
  );
}
