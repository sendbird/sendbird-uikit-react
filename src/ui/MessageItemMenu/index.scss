@import '../../styles/variables';

.sendbird-message-item-menu {
  .sendbird-message-item-menu__trigger {
    .sendbird-message-item-menu__trigger__icon {

    }
  }
}

.sendbird-message-item-menu__list {
  .sendbird-message-item-menu__list__menu-item {

  }
}
