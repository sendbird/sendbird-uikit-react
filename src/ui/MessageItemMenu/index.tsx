import React, { ReactElement, useContext, useRef } from 'react';
import { FileMessage, GroupChannel, OpenChannel, UserMessage } from 'sendbird';

import ContextMenu, { MenuItems, MenuItem } from '../ContextMenu';
import Icon, { IconTypes, IconColors } from '../Icon';
import IconButton from '../IconButton';
import {
  getClassName,
  copyToClipboard,
  isUserMessage,
  isSentMessage,
  isFailedMessage,
  isPendingMessage,
} from '../../utils/index';
import { LocalizationContext } from '../../lib/LocalizationContext';

interface Props {
  className?: string | Array<string>;
  message: UserMessage | FileMessage;
  channel: GroupChannel | OpenChannel;
  isByMe?: boolean;
  disabled?: boolean;
  showEdit?: (bool: boolean) => void;
  showRemove?: (bool: boolean) => void;
  resendMessage?: (message: UserMessage | FileMessage) => void;
  setSupposedHover?: (bool: boolean) => void;
}

export default function MessageItemMenu({
  className,
  message,
  channel,
  isByMe = false,
  disabled = false,
  showEdit,
  showRemove,
  resendMessage,
  setSupposedHover,
}: Props): ReactElement {
  const { stringSet } = useContext(LocalizationContext);
  const triggerRef = useRef(null);
  const containerRef = useRef(null);

  const showMenuItemCopy: boolean = isUserMessage(message as UserMessage);
  const showMenuItemReply: boolean = false && !isFailedMessage(channel, message) && !isPendingMessage(channel, message);
  const showMenuItemEdit: boolean = (isUserMessage(message as UserMessage) && isSentMessage(channel, message) && isByMe);
  const showMenuItemResend: boolean = (isFailedMessage(channel, message) && message.isResendable() && isByMe);
  const showMenuItemDelete: boolean = (isSentMessage(channel, message) && isByMe);

  if (!(showMenuItemCopy || showMenuItemEdit || showMenuItemResend || showMenuItemDelete)) {
    return null;
  }
  return (
    <div
      className={getClassName([className, 'sendbird-message-item-menu'])}
      ref={containerRef}
    >
      <ContextMenu
        menuTrigger={(toggleDropdown: () => void): ReactElement => (
          <IconButton
            className="sendbird-message-item-menu__trigger"
            ref={triggerRef}
            width="32px"
            height="32px"
            onClick={(): void => {
              toggleDropdown();
              setSupposedHover(true);
            }}
            onBlur={(): void => {
              setSupposedHover(false);
            }}
          >
            <Icon
              className="sendbird-message-item-menu__trigger__icon"
              type={IconTypes.MORE}
              fillColor={IconColors.CONTENT_INVERSE}
              width="24px"
              height="24px"
            />
          </IconButton>
        )}
        menuItems={(close: () => void): ReactElement => {
          const closeDropdown = (): void => {
            close();
            setSupposedHover(false);
          };
          return (
            <MenuItems
              className="sendbird-message-item-menu__list"
              parentRef={triggerRef}
              parentContainRef={containerRef}
              closeDropdown={closeDropdown}
              openLeft={isByMe}
            >
              {showMenuItemCopy && (
                <MenuItem
                  className="sendbird-message-item-menu__list__menu-item"
                  onClick={() => {
                    copyToClipboard((message as UserMessage)?.message);
                    closeDropdown();
                  }}
                >
                  {stringSet.MESSAGE_MENU__COPY}
                </MenuItem>
              )}
              {showMenuItemReply && (
                <MenuItem
                  className="sendbird-message-item-menu__list__menu-item"
                  onClick={() => {
                    // TODO: Add replying message logic
                    closeDropdown();
                  }}
                  disable={message?.parentMessageId > 0}
                >
                  {stringSet.MESSAGE_MENU__REPLY}
                </MenuItem>
              )}
              {showMenuItemEdit && (
                <MenuItem
                  className="sendbird-message-item-menu__list__menu-item"
                  onClick={() => {
                    if (!disabled) {
                      showEdit(true);
                      closeDropdown();
                    }
                  }}
                >
                  {stringSet.MESSAGE_MENU__EDIT}
                </MenuItem>
              )}
              {showMenuItemResend && (
                <MenuItem
                  className="sendbird-message-item-menu__list__menu-item"
                  onClick={() => {
                    if (!disabled) {
                      resendMessage(message);
                      closeDropdown();
                    }
                  }}
                >
                  {stringSet.MESSAGE_MENU__RESEND}
                </MenuItem>
              )}
              {showMenuItemDelete && (
                <MenuItem
                  className="sendbird-message-item-menu__list__menu-item"
                  onClick={() => {
                    if (!disabled) {
                      showRemove(true);
                      closeDropdown();
                    }
                  }}
                  disable={message?.threadInfo?.replyCount > 0}
                >
                  {stringSet.MESSAGE_MENU__DELETE}
                </MenuItem>
              )}
            </MenuItems>
          );
        }}
      />
    </div>
  );
}
