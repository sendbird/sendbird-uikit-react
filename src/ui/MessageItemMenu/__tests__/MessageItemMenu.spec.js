import React from 'react';
import { render, screen } from '@testing-library/react';

import MessageItemMenu from "../index";
import { MenuRoot } from '../../ContextMenu';

const createMockChannel = (process) => {
  const mockChannel = {
    isGroupChannel: () => true,
    getUnreadMemberCount: () => 1,
    getUndeliveredMemberCount: () => 1,
  };
  return process ? process(mockChannel) : mockChannel;
};
const createMockMessage = (process) => {
  const mockMessage = {
    messageId: 1010,
    messageType: 'user',
    message: 'First second third',
    url: '',
    name: '',
    type: '',
    thumbnails: [],
    reactions: [],
    sendingStatus: 'succeeded',
    parentMessageId: 0,
    parentMessageInfo: null,
    sender: {
      profileUrl: '',
      userId: 'user-id-001',
      nickname: 'Mathew',
    },
    threadInfo: {
      replyCount: 0,
    },
    isAdminMessage: () => false,
    isUserMessage: () => true,
    isFileMessage: () => false,
    isResendable: () => false,
  };
  return process ? process(mockMessage) : mockMessage;
};

describe('ui/MessageItemMenu', () => {
  it('should have components by own basic status', () => {
    const className="class-name-for-test";
    const { container } = render(
      <div>
        <MenuRoot />
        <MessageItemMenu
          className={className}
          message={createMockMessage()}
          channel={createMockChannel()}
          setSupposedHover={() => {
            console.log('무야호')
          }}
        />
      </div>
    );
    expect(
      container.getElementsByClassName('sendbird-dropdown-portal')[0].id
    ).toBe('sendbird-dropdown-portal');
    expect(
      container.getElementsByClassName('sendbird-message-item-menu').length
    ).toBe(1);
    expect(
      container.querySelector(`.${className}.sendbird-message-item-menu`)
    ).toBeTruthy();
    expect(
      container.getElementsByClassName('sendbird-message-item-menu__trigger').length
    ).toBe(1);
    expect(
      container.getElementsByClassName('sendbird-message-item-menu__trigger__icon').length
    ).toBe(1);
  });

  // TODO: Add tests with onClick events

  it('should do a snapshot test of the MessageItemMenu DOM', function() {
    const { asFragment } = render(
      <MessageItemMenu
        className="classname-for-snapshot"
        message={createMockMessage()}
        channel={createMockChannel()}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
