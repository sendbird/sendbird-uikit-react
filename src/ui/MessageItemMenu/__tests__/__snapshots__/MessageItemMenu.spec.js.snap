// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItemMenu should do a snapshot test of the MessageItemMenu DOM 1`] = `
<div
  className="classname-for-snapshot sendbird-message-item-menu"
>
  <div
    className="sendbird-context-menu"
  >
    <button
      className="sendbird-message-item-menu__trigger sendbird-iconbutton "
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      style={
        Object {
          "height": "32px",
          "width": "32px",
        }
      }
      type="button"
    >
      <span
        className="sendbird-iconbutton__inner"
      >
        <div
          className="sendbird-message-item-menu__trigger__icon sendbird-icon sendbird-icon-more sendbird-icon-color--content-inverse"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          style={
            Object {
              "height": "24px",
              "minHeight": "24px",
              "minWidth": "24px",
              "width": "24px",
            }
          }
          tabIndex="0"
        >
          <test-file-stub />
        </div>
      </span>
    </button>
  </div>
</div>
`;
